components/input/input-wrapper.component.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { FocusMonitor } from '@angular/cdk/a11y';
+import { Directionality } from '@angular/cdk/bidi';
+import { NgTemplateOutlet } from '@angular/common';
+import {
+  afterNextRender,
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  contentChild,
+  DestroyRef,
+  ElementRef,
+  forwardRef,
+  inject,
+  signal,
+  ViewEncapsulation
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { NzFormItemFeedbackIconComponent } from 'ng-zorro-antd/core/form';
+import { getStatusClassNames, getVariantClassNames } from 'ng-zorro-antd/core/util';
+import { NzIconModule } from 'ng-zorro-antd/icon';
+import { NZ_SPACE_COMPACT_ITEM_TYPE, NZ_SPACE_COMPACT_SIZE, NzSpaceCompactItemDirective } from 'ng-zorro-antd/space';
+
+import { NzInputAddonAfterDirective, NzInputAddonBeforeDirective } from './input-addon.directive';
+import { NzInputPrefixDirective, NzInputSuffixDirective } from './input-affix.directive';
+import { NzInputDirective } from './input.directive';
+import { NZ_INPUT_WRAPPER } from './tokens';
+
+@Component({
+  selector: 'nz-input-wrapper',
+  exportAs: 'nzInputWrapper',
+  imports: [NzIconModule, NzFormItemFeedbackIconComponent, NgTemplateOutlet],
+  template: `
+    @if (hasAddon()) {
+      <ng-template [ngTemplateOutlet]="inputWithAddonInner" />
+    } @else if (hasAffix()) {
+      <ng-template [ngTemplateOutlet]="inputWithAffixInner" />
+    } @else {
+      <ng-template [ngTemplateOutlet]="input" />
+    }
+
+    <ng-template #inputWithAddonInner>
+      <span class="ant-input-wrapper ant-input-group">
+        @if (addonBefore()) {
+          <span class="ant-input-group-addon">
+            <ng-content select="[nzInputAddonBefore]"></ng-content>
+          </span>
+        }
+
+        @if (hasAffix()) {
+          <ng-template [ngTemplateOutlet]="inputWithAffix" />
+        } @else {
+          <ng-template [ngTemplateOutlet]="input" />
+        }
+
+        @if (addonAfter()) {
+          <span class="ant-input-group-addon">
+            <ng-content select="[nzInputAddonAfter]"></ng-content>
```

Comment:
question: why not using the auto closing tag here? It's a an angular directive finally, normally auto closing will works